Src/Core/Common/ExpressionExtensions.cs

```diff
@@ -129,9 +131,17 @@ public static class ExpressionExtensions
                         node = null;
                         break;
 
+                    case ExpressionType.Call:
+                        var methodCallExpression = node as MethodCallExpression;
+                        if (methodCallExpression?.Method.Name != "get_Item" || methodCallExpression.Arguments.Count != 1 || !(methodCallExpression.Arguments[0] is ConstantExpression))
+                            throw new ArgumentException(unsupportedExpressionMessage);
```

Comment:
Why don't you inline the actual exception message here? 

By the way, I prefer to always use braces, per [these](https://github.com/dennisdoomen/CSharpGuidelines/blob/master/Src/Guidelines/2400_LayoutGuidelines.md) coding conventions